src/guides/optimism-bridge-widget/index.md

```diff
@@ -0,0 +1,1877 @@
+
+## Build a Marketplace DApp on Optimism
+
+Now that we've got the bridging basics down, let's use the NFT marketplace tutorial in this guide and add a new page specifically for users to bridge their ETH. We will be organizing the project folder structure into two parts: a `truffle` folder for all of your smart contract development needs, and a `client` folder to hold your code for the marketplace frontend.
+
+### Move bridge contents into a truffle folder
+
+Now that we are adding a client component to our DApp, let's place all the relevant Truffle files in a separate folder for organization. Create a folder called `truffle`. Then, move the `contracts`, `migrations`, `scripts`, and`node_modules` folders into the `truffle` folder. Then, also move `.env`, `package.json`, `package-lock.json`, `truffle-config.js`, and `truffle-config.ovm.js` into the `truffle folder`.
+
+We don't need the `.env.example` anymore, so delete that too.
+
+Your top level directory folder structure should look like this:
+
+```shell
+.
+├── .gitignore
+├── LICENSE
+├── README.md
+├── client
+└── truffle
+
+```
+
+Your `truffle` folder structure should look like this:
+
+```shell
+truffle
+├── .env
+├── contracts
+├── migrations
+├── node_modules
+├── package-lock.json
+├── package.json
+├── scripts
+├── test
+├── truffle-config.js
+└── truffle-config.ovm.js
+```
+
+We also don't need the `Greeter` contracts, so let's go ahead and delete the related files: `GreeterL1.sol`, `GreeterL2.sol`, `Greeter.sol`, all the migrations files, and `goerli_bridge_message.mjs`.
+
+### Create client folder using Next.js
+
+To set up the frontend, we'll be using Next.js and Tailwind for our CSS. To do so, call:
+
+```shell
+npx create-next-app@latest client
+```
+
+Then, to add Tailwind to your project:
+
+```shell
+cd client
+npm install -D tailwindcss@latest postcss@latest autoprefixer@latest
+npx tailwindcss init -p
+```
+
+Edit `tailwind.config.js`
+
+```javascript
+module.exports = {
+  content: [
+    "./pages/**/*.{js,ts,jsx,tsx}",
+    "./components/**/*.{js,ts,jsx,tsx}",
+  ],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+}
+```
+
+And replace the code from `styles/global.css`
+
+```css
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+
+Nice! Now that we have our folder structure organized, let's first create our marketplace contracts.
+
+### Create an NFT and Marketplace smart contract
+
+In this tutorial, we'll be using the same contracts and frontend as the tutorial [here](https://trufflesuite.com/guides/nft-marketplace/), so if you want any in depth explanation of the smart contract code, please reference [here](https://trufflesuite.com/guides/nft-marketplace/#build-the-nft-smart-contract) and [here](https://trufflesuite.com/guides/nft-marketplace/#build-the-marketplace-contract).
```

Comment:
Rather than "please reference here and here" a better convention would be something like "please reference the 'Build the NFT Smart Contract' and 'Build the Marketplace Contract' sections of our NFT Marketplace Dapp guide." (although it is bit more verbose).